### ISSUE TITLE
Unable to run kustomize build on remote base in gitlab

### ISSUE THREAD

**Issue by papezt** [kubernetes-sigs/kustomize#4093]
Unable to run kustomize on remote base in gitlab. I have tested it with `kubectl kustomize` and it works, with `kustomize build` it throws error, more in the description.

kustomization.yaml

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

bases:
  - https://gitlab-ci-token:<mytoken>@git.mycompany.com/k8s-apps/redis.git
```

redis.yaml is simplified kutomize to service only:

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - service.yaml
```

service.yaml

```apiVersion: v1
kind: Service
metadata:
  name: redis
spec:
  ports:
  - port: 1234
  selector:
    app: redis
  clusterIP: None
```

**Expected output**

Just to see the kustomized file, in this case, the original file. 

**Error message**
```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
sigs.k8s.io/kustomize/api/internal/git.parseGitUrl(0xc0000b2d20, 0x5a, 0xc0004d6400, 0x0, 0x7f64e3b5d318, 0x18, 0x18, 0x7f64e3b53108, 0x18, 0xc00050e0d8, ...)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/git/repospec.go:128 +0x805
sigs.k8s.io/kustomize/api/internal/git.NewRepoSpecFromUrl(0xc0000b2d20, 0x5a, 0xc000555450, 0x9a2c1d, 0xbb6ae0)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/git/repospec.go:89 +0x76
sigs.k8s.io/kustomize/api/loader.(*fileLoader).New(0xc0003660f0, 0xc0000b2d20, 0x5a, 0x5a, 0xbb6b40, 0xc00050e0d8, 0x203000)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/loader/fileloader.go:174 +0x5f
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateResources(0xc000323000, 0xc000390160, 0xc000363300, 0x1, 0x1, 0xf37b60, 0x0, 0xc00029f578)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/target/kusttarget.go:339 +0x147
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).accumulateTarget(0xc000323000, 0xc000390160, 0xc00029f610, 0x7efd13, 0x0)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/target/kusttarget.go:163 +0x74
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).AccumulateTarget(0xc000323000, 0x0, 0xffffffffffffffff, 0x0)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/target/kusttarget.go:156 +0xce
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).makeCustomizedResMap(0xc000323000, 0x0, 0x0, 0x0, 0x1)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/target/kusttarget.go:111 +0x2f
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).MakeCustomizedResMap(...)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/internal/target/kusttarget.go:107
sigs.k8s.io/kustomize/api/krusty.(*Kustomizer).Run(0xc000555d60, 0xbcb760, 0xf66538, 0xae35cf, 0x1, 0x0, 0x0, 0x0, 0x0)
        /go/pkg/mod/sigs.k8s.io/kustomize/api@v0.8.11/krusty/kustomizer.go:88 +0x3dd
sigs.k8s.io/kustomize/kustomize/v4/commands/build.NewCmdBuild.func1(0xc000346580, 0xf66538, 0x0, 0x0, 0x0, 0x0)
        /workspace/myClone/kustomize/commands/build/build.go:80 +0x1c9
github.com/spf13/cobra.(*Command).execute(0xc000346580, 0xf66538, 0x0, 0x0, 0xc000346580, 0xf66538)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000346000, 0x0, 0xa12400, 0xc000070058)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /workspace/myClone/kustomize/main.go:14 +0x2a
``` 

**Kustomize version**

`{Version:kustomize/v4.2.0 GitCommit:d53a2ad45d04b0264bcee9e19879437d851cb778 BuildDate:2021-06-30T22:49:26Z GoOs:linux GoArch:amd64}`

kubectl version which works

`Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:59:11Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}`

**Platform**

Linux + Windows WSL

**Comment by KnVerey** [comment]
/triage accepted

Gitlab URLs aren't currently supported, but we should fail with a more informative error message.

**Comment by hiroyaonoe** [comment]
Hi!
This problem seems to be fixed by #4039 .

However, #4039 has been applied only to v4.3.0 and later, and not v4.2.0.
In [v4.3.0](https://github.com/kubernetes-sigs/kustomize/tree/kustomize/v4.3.0)
https://github.com/kubernetes-sigs/kustomize/blob/cd17338759ef64c14307991fd25d52259697f1fb/api/internal/git/repospec.go#L128
In [v4.2.0](https://github.com/kubernetes-sigs/kustomize/tree/kustomize/v4.2.0)
https://github.com/kubernetes-sigs/kustomize/blob/d53a2ad45d04b0264bcee9e19879437d851cb778/api/internal/git/repospec.go#L128


In v4.3.0, kustomize seems to try to run `git remote add origin https://gitlab-ci-token:<mytoken>@git.mycompany.com/k8s-apps/redis.git` and `git fetch --depth=1 origin` and throw a more informative error of git.
https://github.com/kubernetes-sigs/kustomize/blob/4d002af7356776cf72a4a92ec5eb4d5c7b022b27/api/internal/git/cloner.go#L25-L35
(`repoSpec.CloneSpec()` returns `https://gitlab-ci-token:<mytoken>@git.mycompany.com/k8s-apps/redis.git`.)

How about updating to v4.3.0?

**Comment by Velociraptor85** [comment]
I "resolved" this by adding an `?` after .git
